packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -238,6 +250,93 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         );
     }
 
+    /**
+     * @notice close short position with user Uniswap v3 LP NFT 
+     * @dev user should approve this contract for Uni NFT transfer
+     */
+    function closeShortWithUserNft(
+        CloseShortWithUserNftParams calldata params
+    ) external {
+        INonfungiblePositionManager(nonfungiblePositionManager).safeTransferFrom(msg.sender, address(this), params.tokenId);
+
+        (uint128 liquidity, ,) = ControllerHelperLib._getUniPositionBalances(nonfungiblePositionManager, params.tokenId, IOracle(oracle).getTimeWeightedAverageTickSafe(wPowerPerpPool, 420), isWethToken0);
```

Comment:
Will probably change this with the line commented below, will keep this for now.